Ingress/controllers/nginx-third-party/nginx/utils.go

```diff
@@ -0,0 +1,192 @@
+		if err != nil {
+			glog.Errorf("error queriyng service kube-system/kube-dns: %v", err)
+			return false, nil
+		}
+
+		ipPort = fmt.Sprintf("%v:%v", svc.Spec.ClusterIP, 53)
+		return true, nil
+	})
+
+	return
+}
+
+func getService(kubeClient *unversioned.Client, name string) k8sService {
+	// Wait for the default backend Service. There's no pretty way to do this.
+	parts := strings.Split(name, "/")
+	if len(parts) != 2 {
+		glog.Fatalf("Default backend should take the form namespace/name: %v", name)
+	}
+
+	defaultPort, err := getPort(kubeClient, parts[0], parts[1])
+	if err != nil {
+		glog.Fatalf("Could not configure default backend %v: %v", name, err)
+	}
+
+	return k8sService{
+		ServiceName: parts[1],
+		ServicePort: defaultPort,
+		Namespace:   parts[0],
+	}
+}
+
+func getTcpServices(kubeClient *unversioned.Client, tcpServices string) []k8sService {
+	var svcs []k8sService
+	for _, svc := range strings.Split(tcpServices, ",") {
+		namePort := strings.Split(svc, ":")
+		if len(namePort) == 2 {
+			tcpSvc := getService(kubeClient, namePort[0])
+			tcpSvc.ExposedPort = namePort[1]
+			svcs = append(svcs, tcpSvc)
+		} else {
+			glog.Fatalf("TCP services should take the form namespace/name:port not %v", svc)
+		}
+	}
+
+	return svcs
+}
+
+// from3rdPartyResource obtains the configuration located in the specified
+// namespace with a particular name defined by the env variable CONF_NAME.
+// It returns the last version or error if it doesn't exists
+func cfgFrom3rdPartyResource(kubeClient *unversioned.Client, name string) (nginxConfiguration, error) {
+	namespace := "default"
+	parts := strings.Split(name, "/")
+	confName := parts[0]
+	if len(parts) == 2 {
+		namespace = parts[0]
+		confName = parts[1]
+	}
+
+	path := fmt.Sprintf("/apis/k8s.io/stable/v1/namespaces/%s/nginxingressloadbalancers/%s", namespace, confName)
+	data, err := kubeClient.RESTClient.
+		Get().
+		AbsPath(path).
+		DoRaw()
+
+	if err != nil {
+		glog.Errorf("Failed: %v\nBody: %s", err, string(data))
+		return nginxConfiguration{}, err
+	}
+
+	if glog.V(2) {
+		glog.Infof("Nginx configuration: %s", string(data))
+	}
+
+	var cfg nginxConfiguration
+	err = json.Unmarshal(data, &cfg)
+	if err != nil {
+		glog.Errorf("Invalid json: %v", err)
+		return nginxConfiguration{}, fmt.Errorf("Invalid custom nginx configuration: %v", err)
+	}
+
+	if cfg.SSLDHParam != "" {
+		// we need to dump the content to a file.
+		data, err := base64.StdEncoding.DecodeString(cfg.SSLDHParam)
+		if err != nil {
+			glog.Fatalf("invalid base64 content in parameter SSLDHParam: %v", err)
+		}
+
+		dhDirectory := sslDirectory + "/dh-file-from-cfg"
+		err = os.MkdirAll(dhDirectory, 0644)
+		if err != nil {
+			glog.Fatalf("error creating directory for dh file %q: %v", dhDirectory, err)
+		}
+
+		ioutil.WriteFile(dhDirectory+"/dhparam.pem", data, 0644)
+		cfg.SSLDHParam = ""
+	}
+
+	return cfg, nil
+}
+
+// TODO: is svc.Spec.Port the correct option? why not svc.Spec.NodePort
```

Comment:
why use NodePort? not all services have nodeport right, do you mean for the ones that do? A service can have multiple ports.